src/OneMightyRoar/PHP_Paulus_Components/Environment/EnvironmentFactory.php

```
@@ -0,0 +1,66 @@
+
+use OneMightyRoar\PHP_Paulus_Components\Environment\Environments\CustomEnvironment;
+use OneMightyRoar\PHP_Paulus_Components\Environment\Environments\DevelopmentEnvironment;
+use OneMightyRoar\PHP_Paulus_Components\Environment\Environments\ProductionEnvironment;
+use OneMightyRoar\PHP_Paulus_Components\Environment\Environments\StagingEnvironment;
+
+/**
+ * Class EnvironmentFactory
+ *
+ * A factory class to build an environment class based on the current setup
+ */
+class EnvironmentFactory
+{
+    /**
+     * Constants
+     */
+
+    /**
+     * @type string The default environment to build
+     */
+    const DEFAULT_ENVIRONMENT = DevelopmentEnvironment::ENVIRONMENT_NAME;
+
+
+    /**
+     * Methods
+     */
+
+    /**
+     * Factory create method
+     *
+     * Strictly build an environment based on the passed argument, the current environment, or a default one
+     *
+     * @param string $environment {optional} The specific environment to build
+     * @return CustomEnvironment|DevelopmentEnvironment|ProductionEnvironment|StagingEnvironment
+     */
+    public static function create($environment = null)
+    {
+        // First check if we passed in an environment string, if not try to get the current environment variable
+        $environment_string = $environment ?: getenv('ENVIRONMENT');
+        // If there isn't an environment variable available, fall back to our default environment
+        $environment_string = $environment_string ?: static::DEFAULT_ENVIRONMENT;
+
+        if (DevelopmentEnvironment::ENVIRONMENT_NAME === $environment_string) {
+            $environment = new DevelopmentEnvironment();
+        } elseif (StagingEnvironment::ENVIRONMENT_NAME === $environment_string) {
+            $environment = new StagingEnvironment();
+        } elseif (ProductionEnvironment::ENVIRONMENT_NAME === $environment_string) {
+            $environment = new ProductionEnvironment();
+        } else {
+            $environment = new CustomEnvironment();
```

Hmm, well you could either have an "UnknownEnvironment" (for the null object pattern, like we talked about), you could return `null`, or you could throw an `UnexpectedValueException`. It depends on the implications throughout the app. I'd suggest against just returning `null`.